### ISSUE TITLE
Adding diagbox feature to nicematrix

### ISSUE THREAD

**Issue by Basmah1961** [fpantigny/nicematrix#5]
![image](https://github.com/fpantigny/nicematrix/assets/94080257/4fa8cbe9-9cc0-407d-9133-0a7685832e23)

`tabularray` has a command `\diagbox` which allows for convenient placement of diagonal boxes. It might be good to add this command instead of using `\CodeAfter` and `tikzpicture`

```
\documentclass[11pt, a4paper]{article}

\usepackage{fancyhdr}
\pagestyle{empty}

\usepackage{tikz}

\usepackage{nicematrix}

\usepackage{tabularray}
\UseTblrLibrary{diagbox}

\begin{document}

\begin{NiceTabular}[hvlines, cell-space-limits=0pt]{l l l}

	
	& Beta
	& Gamma
	\\
	
	Epsilon
	& Zeta
	& Eta
	\\
	
	Iota
	& Kappa
	& Lambda
	\\
	
	\CodeAfter
	\begin{tikzpicture}[remember picture,overlay,outer sep=0pt,inner sep=0pt]
		
		\draw[] (1-|1)--(2-|2);
		
		\node[anchor=north east] at ([shift={(-1pt,-1pt)}] 1-|2) {Pp};
		
		\node[anchor=south west] at ([shift={(+1pt,+1pt)}] 2-|1) {Aa};
		
	\end{tikzpicture}
	
\end{NiceTabular}

\bigskip

\begin{tblr}{colspec={Q[l,h] Q[l,t] Q[l,h]},
		rowsep=0pt, 
		hlines,vlines
}
	
	\diagbox{Aa}{Pp}
	& Beta
	& Gamma
	\\
	
	Epsilon
	& Zeta
	& Eta
	\\
	
	Iota
	& Kappa
	& Lambda
	\\
	
\end{tblr}

\end{document}
```

**Comment by fpantigny** [comment]
`nicematrix` yet has such feature with its built-in command `\diagbox`.

```
\documentclass[11pt, a4paper]{article}
\pagestyle{empty}
\usepackage{nicematrix}

\begin{document}

\begin{NiceTabular}[hvlines]{l l l}
\diagbox{Aa}{Pp}
& Beta
& Gamma
\\

Epsilon
& Zeta
& Eta
\\

Iota
& Kappa
& Lambda
\\
\end{NiceTabular}

\end{document}
```

![image](https://github.com/fpantigny/nicematrix/assets/39270854/ae03f336-e637-45b8-bf84-4e3cbbc34ecd)

**Comment by Basmah1961** [comment]
> `nicematrix` yet has such feature with its built-in command `\diagbox`.
> 
> ```
> \documentclass[11pt, a4paper]{article}
> \pagestyle{empty}
> \usepackage{nicematrix}
> 
> \begin{document}
> 
> \begin{NiceTabular}[hvlines]{l l l}
> \diagbox{Aa}{Pp}
> & Beta
> & Gamma
> \\
> 
> Epsilon
> & Zeta
> & Eta
> \\
> 
> Iota
> & Kappa
> & Lambda
> \\
> \end{NiceTabular}
> 
> \end{document}
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/39270854/346106313-ae03f336-e637-45b8-bf84-4e3cbbc34ecd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNzg1NjgsIm5iZiI6MTcyMDE3ODI2OCwicGF0aCI6Ii8zOTI3MDg1NC8zNDYxMDYzMTMtYWUwM2YzMzYtZTYzNy00NWI4LWJmODQtNGUzY2JiYzM0ZWNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA1VDExMTc0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk0NjgzYzg5Yzg5M2I5NTk5OTNhNTU5NTI2MTlmMmM4YzM0ODUzN2M4MDgwYzBiNDUxYmQ5NGFhYmI5MDRhOGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SUu8iuyJdbW0iJ2xhlPNnHUR4wtpczxBYVrJhvNTsjE)

I did not know it available. However, I think it is possible to improve it by adding horizontal white space. To improve readability. Also, `Aa` aligns to the first column cells and `Pb` aligns to the left of the first column with a space of `tabcolsep` between it and the vertical rule. This is the expected result

**Comment by fpantigny** [comment]
You are right. It's a good suggestion. However, I can't change easily such features without silently breaking existing documents. The command `\diagbox` has been introduced in the version 4.1 of 2020/05/27 (four years ago) and it's not so difficult for the final user to add a `\hspace*{...}` and a `\strut` (for the vertical space).

**Comment by Basmah1961** [comment]
Then why not make `diagbox` behave as normally does, but add an optional key to change its behavior? This way, existing documents will not be affected while users get to test new functionalities which emulate `tabularray`
